### ISSUE TITLE
Content-specific templates: Offer a way to remap the associated content

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#48645]
After creating a template for a specific entity (e.g. Category: Recipes, Page: About, Post: Hello World), there's no way to change that associated entity afterwards. If you want to do this then you need to copy the blocks in the original template before deleting it, then create a new template and paste those blocks in... obviously this is unnecessarily tedious. 

It's also problematic if the original entity is deleted – you're left with an orphaned template that does nothing. It would be nice if you could easily repurpose that template. 

Such an affordance may even house related actions, for example cloning; "I want to apply the Category: Recipes template to my Ingredients category as well".

The Inspector and Details panel in the dark sidebar seem like they might be good homes for this, but we should explore options.

**Comment by youknowriad** [comment]
I think this is a duplicate of #48247

**Comment by jameskoster** [comment]
Not quite. I think #48247 is about making a `$custom` template the default for a given post type. 

This one is about updating a template like `page-$slug` to re-assign which page it applies to.

**Comment by youknowriad** [comment]
For me it's just the same thing, regardless of whether the root template is "$custom" or "page-$slug". You should be able to change the purpose (slug) of the template entirely. That said, I'm fine with the issue being kept open.
